### ISSUE TITLE
Deprecated services docs sounds missing...

### ISSUE THREAD

**Issue by mickaelandrieu** [symfony/symfony-docs#6919]
I don't see any reference on the `deprecated` attribute available when we declare services in docs...

In pdf (before the generation was broken), we had something like:
![deprecating_services](https://cloud.githubusercontent.com/assets/1247388/18004308/591cb930-6b93-11e6-8ca8-a23f11ee85ba.png)

The docs also say:

> The ability to "un-deprecate" a service is possible only when declaring the definition in PHP.

I'd like to see an example, I can't retrieve any reference in docs too.

Mickaël

**Comment by wouterj** [comment]
You mean http://symfony.com/doc/current/service_container/alias_private.html#deprecating-services ?

**Comment by mickaelandrieu** [comment]
Yes (thank you, I don't know how to do to find something in website, I want your technique!).

 I don't realy understand the ability to "un-deprecate": I mean, in PHP you could do:

```php
$container
    ->register('acme.my_service', '...')
    ->setDeprecated(
        false, // so, it's undeprecated, right ?
        'The "%service_id%" service is deprecated since 2.8 and will be removed in 3.0.'
    )
;
```

What is the interest to do it instead of just remove the ``setDeprecated()`` call ?

**Comment by wouterj** [comment]
> What is the interest to do it instead of just remove the setDeprecated() call ?

I think to be able to undeprecate services in a compiler pass or DI extension. To be honest, I don't really understand why you want it either.

 > Yes (thank you, I don't know how to do to find something in website: I want your technique!).

It's mostly just because I saw a PR writing/tweaking the docs about it, so I know where to look for. (knowing where to find stuff is easier than knowing the stuff)

**Comment by mickaelandrieu** [comment]
> I think to be able to undeprecate services in a compiler pass or DI extension. 

True, pretty much an edge case.

Thank you for help, it's more clear to me. Don't you think we may adapt your sentence in docs ? Something like:

> In some cases, you may need to "un-deprecate" services, you can do it in PHP, for instance using a Compiler Pass:

```php
$definition = $container->get('acme.my_service');
$definition->setDeprecated(false);
``` 

Mickaël

**Comment by javiereguiluz** [comment]
I'm not sure we want to document this. @mickaelandrieu your example works perfectly ... but this "feature" is probably something we don't want to promote, so I'd vote to not document it.

**Comment by xabbuh** [comment]
I do not see much value for this "feature". Imo we should simply remove the caution.

**Comment by xabbuh** [comment]
I would be happy to merge it. :)

**Comment by javiereguiluz** [comment]
Closing it as fixed by #7077.
